### PR TITLE
RCON broadcast with spaces

### PR DESCRIPTION
**Pull request by Callum027**
This incorporates the ideas from https://github.com/jammsen/docker-palworld-dedicated-server/pull/243, but updates the existing `rconcli` script to perform the conversion to non-breaking spaces, instead of using a customised `rcon` command.

To make the implementation nicer, in `rconcli` the command and any additional values have been split into separate arguments. The first argument to the script is the desired command (e.g. `broadcast`), and any following arguments will be value (e.g. `Saving done`).

The result (from the `rcon` command's perspective) will be:

```
broadcast Saving<nbsp>done
```

The following changes are also included (some from the above PR):

* Convert all RCON broadcasts to use spaces, instead of dashes, and make them otherwise look nicer.
* Do not remove spaces from player names in broadcasts, as these are now handled correctly.
* Add timestamps to all RCON broadcasts, to make it easier to tell when a broadcast was made.
* When using the RCON `showplayers` command, redirect `stderr` to `/dev/null` to prevent log errors caused by players with special characters in their names.
* Fix typos.
* Remove trailing spaces on lines in some files.

### CODE REVIEW

**Comment by Callum027** [comment]
Broadcast as shown ingame:

image

File: includes/rcon.sh [hunk 1]
```diff
@@ -1,57 +1,56 @@
...
-    rconcli 'save'
-    rconcli 'broadcast Done-saving-server-shuts-down-now'
+    rconcli broadcast "$(get_time) Server shutdown requested. Saving..."
+    rconcli save
+    rconcli broadcast "$(get_time) Saving done. Server shutting down..."
 }
 
 function broadcast_automatic_restart() {
-    time=$(date '+%H:%M:%S')
+    time=
```
Comment (jammsen) [inline includes/rcon.sh:14]: This variable is empty and not used anymore

File: includes/playerdetection.sh [hunk 2]
```diff
@@ -9,19 +9,19 @@ player_detection_loop() {
...
         sleep "$RCON_PLAYER_DETECTION_CHECK_INTERVAL"
-    done 
+    done
 }
 
 # Function to compare current and previous player lists
 compare_players() {
     local old_players=("${current_players[@]}")
-    readarray -t current_players < <(rcon showplayers | tail -n +2)
+    readarray -t current_players < <(rcon showplayers 2> /dev/null | tail -n +2)
```
Comment (jammsen) [inline includes/playerdetection.sh:18]: Why is this needed?

File: scripts/restart.sh [hunk 3]
```diff
@@ -22,23 +22,22 @@ function schedule_restart() {
                     send_restart_now_notification
                 fi
                 break
-            else 
+            else
                 ew ">>> Server has still players"
             fi
-            time=$(date '+%H:%M:%S')
-            rconcli "broadcast ${time}-AUTOMATIC-RESTART-IN-$counter-MINUTES"
+            rconcli broadcast "$(get_time) AUTOMATIC RESTART IN $counter MINUTES"
```
Comment (jammsen) [inline scripts/restart.sh:28]: function get_time() is defined in rcon.sh
The restart-script doesnt include a source of this file, so this will never be reached.

File: includes/rcon.sh [hunk 4]
```diff
@@ -1,57 +1,56 @@
 # shellcheck disable=SC2148
 
+function get_time() {
```
Comment (jammsen) [inline includes/rcon.sh:3]: Why is this a function that gets called multiple times and always has to calculate the date? Why not defined once, use once the resources and save it to memory. What am i missing? Why is this better?

**Review by jammsen — CHANGES_REQUESTED** [review]

**Comment by jammsen** [comment]
Im working now on this, but its only part1 of an update that needs to be done to be current again. @Callum027

File: includes/rcon.sh (hunk 1 above)
Comment (Callum027) [reply includes/rcon.sh]: Thanks, removed the line.

File: scripts/restart.sh (hunk 3 above)
Comment (Callum027) [reply scripts/restart.sh:29]: Sourced `/includes/rcon.sh` at the top of the file.

File: includes/rcon.sh (hunk 4 above)
Comment (Callum027) [reply includes/rcon.sh:3]: This is run every time for each line because each line is broadcasted at a different time, and each line needs to have an accurate timestamp for when it was broadcasted.

Take backups for example. It takes 10+ seconds to run, so calculating unique timestamps for each broadcast will accurately show when the it starts and ends when broadcasted in the game.
